biomaj/bank.py

```diff
@@ -525,10 +527,36 @@ def save_session(self):
 
         self.bank = self.banks.find_one({'name': self.name})
 
+    def format_release(self, release, reverse=False):
+        """
+        Check the release does not contains any dot '.' character that should make MongoDB complain
+
+        :param release: release number to check
+        :type release: str
+        :param reverse: Do the reverse replacement
+        :type reverse: bool
+        :return: Release string with dot '.' replaced with character set in config (release.replace)
+        :rtype: str
+        """
+        if not self.config.get('release.replacedot'):
```

Comment:
therer should be a single place (or like) for this in workflow.py, I gonna
have a look at the code

Le mar. 26 avr. 2016 à 17:46, Emmanuel Quevillon <notifications@github.com>
a écrit :

> In biomaj/bank.py
> <https://github.com/genouest/biomaj/pull/44#discussion_r61110671>:
>
> > @@ -525,10 +527,36 @@ def save_session(self):
> >
> >          self.bank = self.banks.find_one({'name': self.name})
> >
> > +    def format_release(self, release, reverse=False):
> > +        """
> > +        Check the release does not contains any dot '.' character that should make MongoDB complain
> > +
> > +        :param release: release number to check
> > +        :type release: str
> > +        :param reverse: Do the reverse replacement
> > +        :type reverse: bool
> > +        :return: Release string with dot '.' replaced with character set in config (release.replace)
> > +        :rtype: str
> > +        """
> > +        if not self.config.get('release.replacedot'):
>
> yes I always forget about default value available as argument.
> I've check in workflow.py to directly set/replace the release using
> format_release.
> It also looks that setting of remote release is done in multiple different
> places:
>
> config.py:279:        self.set('remoterelease', '')
> workflow.py:68:        self.session.config.set('remoterelease', '')
> workflow.py:401:                    self.session.set('remoterelease', dep.session.get('remoterelease'))
> workflow.py:517:        self.session.set('remoterelease', release)
> workflow.py:718:            self.session.config.set('remoterelease', self.session.get('remoterelease'))
> workflow.py:871:            self.session.set('remoterelease', release)
> workflow.py:891:        self.session.config.set('remoterelease', self.session.get('remoterelease'))
> workflow.py:1253:                    session.set('remoterelease', prod['remoterelease'])
>
> I'll try to move this modification into workflow.py
> The thing is we will need a method accessible from other modules and also
> able to access the config object.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/genouest/biomaj/pull/44/files/d235f544d677e718229ab04b92229485fa50ca28#r61110671>
>